### PR TITLE
Rf/autoindentpaste improvements

### PR DESCRIPTION
**Pull request by oheil**
Improvements based on rf/autoindentpaste
https://github.com/JuliaLang/julia/tree/rf/autoindentpaste

For additional issues which are solved by this PR see
https://discourse.julialang.org/t/repl-copy-paste-and-auto-indent-in-julia-0-7/10230
e.g. pasting a mix of space and tabs and empty lines at once still break autoindentation

This PR replaces the PR of @rfourquet:
https://github.com/JuliaLang/julia/pull/29129

### CODE REVIEW

**Comment by oheil** [comment]
pain... I will check for this :-(

File: stdlib/REPL/src/LineEdit.jl [hunk 1]
```diff
@@ -680,6 +682,27 @@ edit_splice!(s, ins::AbstractString) = edit_splice!(s, region(s), ins)
 function edit_insert(s::PromptState, c)
     push_undo(s)
     buf = s.input_buffer
+
+    if ! options(s).auto_indent_bracketed_paste
```
Comment (musm) [inline stdlib/REPL/src/LineEdit.jl]: `if !options(s).auto_indent_bracketed_paste`

File: stdlib/REPL/src/LineEdit.jl [hunk 2]
```diff
@@ -680,6 +682,27 @@ edit_splice!(s, ins::AbstractString) = edit_splice!(s, region(s), ins)
...
+        pos=position(buf)
```
Comment (musm) [inline stdlib/REPL/src/LineEdit.jl]: `pos = position(buf)`

**Comment by musm** [comment]
bump, this is an annoying issue when using the terminal.

**Comment by KristofferC** [comment]
Yeah, I was planning on trying this out and merging this. Will do so now.

**Comment by musm** [comment]
I just tested this out, situation is improved but it's still not pasting everything in properly

**Comment by musm** [comment]
E.g. sometimes it pastes and indents it correctly other times it does whatever it wants to...

**Comment by musm** [comment]
I haven't tested the original PR that this is based on, however.

**Comment by oheil** [comment]
> E.g. sometimes it pastes and indents it correctly other times it does whatever it wants to...

Can you give some examples? Maybe there are still cases which fail.
It could also be a timing problem, because detection of copy&paste is based on a time measurement.

**Comment by musm** [comment]
@oheil  All I did was copy some code and repeatedly paste it into the terminal.
The first time it didn't indent things correctly, but then it would sporadically paste it correctly and also incorrectly.

**Comment by oheil** [comment]
I guess it could be the timing. If you like to, you may try out other thresholds, using e.g.
`Base.active_repl.options.auto_indent_time_threshold = 0.01`
the current value is 0.005. The higher the value the higher is the chance that auto_indent is detected. If it is too high, normal keyboard input could be missinterpreted as copy&past and than auto_indent would be bad.

**Comment by KristofferC** [comment]
I think we should try this for a while, it doesn't seem any worse locally. If it leads to problems we can revert but things like this pretty much have to be tested in the wild.

**Pull request merged**